### PR TITLE
Adds a noisy deletion timer to autosay mobs

### PR DESCRIPTION
**Pull request by Crazylemon64**
This is to satisfy @Fox-McCloud - this way, it doesn't fail silently, and we have a rough idea of when it's going on.

### CODE REVIEW

File: code/game/objects/items/devices/radio/radio.dm [hunk 1]
```diff
@@ -262,8 +262,22 @@ var/global/list/default_medbay_channels = list(
...
+	lifetime_timer = addtimer(src, "autocleanup", SecondsToTicks(10))
+	..()
+
+/mob/living/automatedannouncer/Destroy()
+	if(lifetime_timer)
+		deltimer(lifetime_timer)
+	..()
+
+/mob/living/automatedannouncer/proc/autocleanup()
+	log_debug("An announcer somehow managed to outlive the radio! Deleting! Area: [get_area(src)], Loc: [src.locs[1]]"
```
Comment (Crazylemon64) [inline code/game/objects/items/devices/radio/radio.dm]: Wait this is dumb hold on

File: code/game/objects/items/devices/radio/radio.dm [hunk 2]
```diff
@@ -262,8 +262,22 @@ var/global/list/default_medbay_channels = list(
...
+	lifetime_timer = addtimer(src, "autocleanup", SecondsToTicks(10))
+	..()
+
+/mob/living/automatedannouncer/Destroy()
+	if(lifetime_timer)
+		deltimer(lifetime_timer)
+	..()
+
+/mob/living/automatedannouncer/proc/autocleanup()
+	log_debug("An announcer somehow managed to outlive the radio! Deleting! Area: [get_area(src)], Loc: [src.loc.loc]"
```
Comment (Crazylemon64) [inline code/game/objects/items/devices/radio/radio.dm]: Somehow I get the idea this is just another possible runtime again. Goodbye, loc term

**Comment by Crazylemon64** [comment]
Wait why doesn't this compile did I do something dumb

**Comment by Fox-McCloud** [comment]
Ah, thanks, though my comment was more of a joke on how no matter how many times we fix something, invariable, something goes wrong somewhere else.

**Comment by Tastyfish** [comment]
Might be useful if the log mesage also had the autosay message and name.

**Comment by Crazylemon64** [comment]
Got it

**Pull request merged**